geonode/base/models.py

```diff
@@ -337,76 +338,101 @@ def get_queryset(self):
 
 class HierarchicalKeyword(TagBase, MP_Node):
     node_order_by = ['name']
-
     objects = HierarchicalKeywordManager()
 
     @classmethod
-    def dump_bulk_tree(cls, user, parent=None, keep_ids=True, type=None):
-        """Dumps a tree branch to a python data structure."""
+    def resource_keywords_tree(cls, user, parent=None, resource_type=None, resource_name=None):
+        """ Returns resource keywords tree as a dict object. """
         user = user or get_anonymous_user()
-        ctype_filter = [type, ] if type else ['layer', 'map', 'document']
-        qset = cls._get_serializable_model().get_tree(parent)
+        resource_types = [resource_type] if resource_type else ['layer', 'map', 'document']
```

Comment:
We need to add here also the `GeoApp` subtypes in the case the `resource_type` was not valorized.

Add the `apps subtypes` other than `['layer', 'map', 'document']` to the default `ctypes`

Reply:
I just continued it with the existing code input. As you guys suggest.

Reply:
If possible and not too hard, it would be good to improve this part.